src/relay/op/type_relations.cc

```diff
@@ -126,6 +126,7 @@ bool BroadcastCompRel(const Array<Type>& types, int num_inputs, const Attrs& att
       return true;
     }
   }
+  reporter->Assign(types[0], types[1]);
```

Comment:
why?

Reply:
```
    # f(x) = if x > 0 then g(x - 1) else 0
    # g(y) = if y > 0 then f(y - 1) else 0
```
this test doesn't typecheck otherwise

Reply:
even if you add explicit type? not doing anything is the correct thing, and i dont think you should do this (look unsound to me) to make the type checker stronger - probably too strong.

Reply:
removed, added type annotation to test. thanks!